src/scripts/main.js

```diff
@@ -1,3 +1,76 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const appendRowButton = document.querySelector('.append-row');
+  const removeRowButton = document.querySelector('.remove-row');
+  const appendColumnButton = document.querySelector('.append-column');
+  const removeColumnButton = document.querySelector('.remove-column');
+
+  const table = document.querySelector('.field');
+
+  const MAX_ROWS = 10;
+  const MIN_ROWS = 2;
+  const MAX_COLUMNS = 10;
+  const MIN_COLUMNS = 2;
+
+  const updateButtonsState = () => {
+    const rowCount = table.rows.length;
+    const columnCount = table.rows[0]?.cells.length || 0;
```

Comment:
Ensure that the table has at least one row before accessing `table.rows[0]`. If the table is initially empty, this could cause an error. Consider adding a check to handle this scenario.